### PR TITLE
Update dimensions when the remote track is detached

### PR DESCRIPTION
**Pull request by burzomir**
@davidzhao @lukasIO 

Use case for this change:
There are two video elements in the UI. The first is small (480x320) and placed in a participant grid. The second is big (1280x720) and only displays the active speaker. When no one is speaking, then the second video element is removed. In this case, I want to use only the low-resolution simulcast layer. Currently, simulcast layers change only on resize and visibility change events or when a remote track is attached.

### CODE REVIEW

**Comment by lukasIO** [comment]
hi @burzomir ,
thanks for the PR, this sounds like a reasonable change. 

Instead of calling it in `detach` could you add it to `stopObservingElementInfo` instead? We have `updateVisibility` there already.

**Comment by burzomir** [comment]
@lukasIO Do you want me to replace `this.updateDimensions();` with `this.stopObservingElementInfo(elementInfo)` in this code?

```
detach(): HTMLMediaElement[];
  detach(element: HTMLMediaElement): HTMLMediaElement;
  detach(element?: HTMLMediaElement): HTMLMediaElement | HTMLMediaElement[] {
    let detachedElements: HTMLMediaElement[] = [];
    if (element) {
      this.stopObservingElement(element);
      this.updateDimensions();
      return super.detach(element);
    }
    detachedElements = super.detach();

    for (const e of detachedElements) {
      this.stopObservingElement(e);
    }
    this.updateDimensions();

    return detachedElements;
  }
  ```

File: src/room/track/RemoteVideoTrack.ts [hunk 1]
```diff
@@ -123,6 +123,7 @@ export default class RemoteVideoTrack extends RemoteTrack {
     }
     this.elementInfos = this.elementInfos.filter((info) => info !== elementInfo);
     this.updateVisibility();
+    this.updateDimensions();
```
Comment (lukasIO) [inline src/room/track/RemoteVideoTrack.ts:126]: I guess especially because we're calling this in a loop, it would make sense to replace this with the debounced version
```suggestion
    this.debouncedHandleResize();
```

Comment (burzomir) [reply src/room/track/RemoteVideoTrack.ts]: @lukasIO Done.

**Review by davidliu — APPROVED** [review]
One small comment, but LGTM.

File: src/room/track/RemoteVideoTrack.ts [hunk 2]
```diff
@@ -195,7 +196,7 @@ export default class RemoteVideoTrack extends RemoteTrack {
   private stopObservingElement(element: HTMLMediaElement) {
     const stopElementInfos = this.elementInfos.filter((info) => info.element === element);
     for (const info of stopElementInfos) {
-      info.stopObserving();
+      this.stopObservingElementInfo(info);
     }
     this.elementInfos = this.elementInfos.filter((info) => info.element !== element);
```
Comment (davidliu) [inline src/room/track/RemoteVideoTrack.ts:201]: I think this line becomes redundant and can be removed?

Comment (burzomir) [reply src/room/track/RemoteVideoTrack.ts]: @davidliu You're right. `elementInfos.filter` is already executed in `stopObservingElementInfo.` I pushed a fix.

**Comment by lukasIO** [comment]
this is looking great, thanks @burzomir !
